### PR TITLE
Fix quickstart guide for bigchaindb_driver

### PR DESCRIPTION
**Pull request by muawiakh**
- Fixes quickstart for a fresh Ubuntu/RHEL setup,
and installs all the relevant dependencies.

### CODE REVIEW

**Comment by muawiakh** [comment]
Fixes #315

File: docs/quickstart.rst [hunk 1]
```diff
@@ -34,20 +35,40 @@ On Ubuntu 14.04 and 16.04, this works:
...
+
+.. code-block:: bash
+
+    $ sudo apt-get install libssl-dev
+
+On Fedora 23 and 24:
+
+.. code-block:: bash
+
+    sudo dnf install openssl-devel
```
Comment (ttmc) [inline docs/quickstart.rst]: $

File: docs/quickstart.rst [hunk 2]
```diff
@@ -34,20 +35,40 @@ On Ubuntu 14.04 and 16.04, this works:
...
+
```
Comment (ttmc) [inline docs/quickstart.rst]: Below, Python 3.5+ now becomes "Dependency 3". The following ones have to be renumbered as well.

File: docs/quickstart.rst [hunk 3]
```diff
@@ -87,7 +116,12 @@ Once you have a recent Python 3 version of ``pip``, you should be able to upgrad
 
 .. code-block:: bash
 
-    pip install --upgrade setuptools
+    $ pip install --upgrade setuptools
+
+    OR
+
+    $ sudo pip3 install --upgrade setuptools
```
Comment (ttmc) [inline docs/quickstart.rst]: Why does this one use sudo but the one above doesn't? Same comment for the next subsection.

**Comment by ttmc** [comment]
Never mind codecov in documentation-only pull requests. Clearly it is wrong.

File: docs/quickstart.rst (hunk 2 above)
Comment (muawiakh) [inline docs/quickstart.rst]: Done

File: docs/quickstart.rst (hunk 3 above)
Comment (muawiakh) [inline docs/quickstart.rst]: Because it didn't work for me without the `sudo` and I was not using the `virtualenv` or `conda`, and running this on base VM. I can change that to `pip3 install --upgrade setuptools` as well to keep it consistent, and if the user encounters a permissions issue they can run it with `sudo` which is pretty straight forward. Same for the next subsection, because with virtualenv, I don't need `sudo` permissions.

**Comment by muawiakh** [comment]
@ttmc because the driver + apt pull the dependencies recursively.

File: docs/quickstart.rst (hunk 3 above)
Comment (ttmc) [inline docs/quickstart.rst]: Okay maybe just remove all the `sudo`

**Comment by ttmc** [comment]
> @ttmc because the driver + apt pull the dependencies recursively.

Okay, so when apt installs `libffi-dev` and `libssl-dev`, it also installs all the other packages. That's nice: only two packages to install.

But why, when we install the `bigchaindb` Python package alone, do we have to install a bunch of other OS-level dependencies? The Python Driver `bigchaindb_driver` package depends on the `bigchaindb` package, so it also depends on all of BigchainDB Server's OS-level dependencies, no?

In short, A depends on B, but A has fewer dependencies than B?

**Comment by muawiakh** [comment]
We only need `libssl-dev` and `libffi-dev` for the server quickstart as well. Nothing more. I will update it.

**Comment by kremalicious** [comment]
So is there a way to have `apt install bigchaindb` automatically install `libssl-dev` and `libffi-dev`? Would be just a bit nicer UX

**Comment by muawiakh** [comment]
@kremalicious No, pip cannot. It can only install python modules as per the `requirements.txt` or `setup.py`
